platform/ios/src/MGLMapView.mm

```diff
@@ -1114,7 +1124,29 @@ - (void)updateFromDisplayLink:(CADisplayLink *)displayLink
         [self updateAnnotationViews];
         [self updateCalloutView];
         
-        [self.glView display];
+#ifdef MGL_RECREATE_GL_IN_AN_EMERGENCY
+#if (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 120200)
```

Comment:
So this #ifdef block is triggered if the user's iOS version is greater than or equal to 12.2, correct? 

Are there more things that need to happen before the #ifdef is triggered?

My first comment on this is that until we have verified that this issue is reproducible on iOS 12.3, let's keep this #ifdef to 12.2 only.

Reply:
`#if` and `#ifdef` are conditional compilation directives that are only evaluated by the compiler at build time, not at runtime. The code below executes if and only if `MGL_RECREATE_GL_IN_AN_EMERGENCY` is defined in the build system and we’re building the SDK against the iOS 12.2 SDK or above (i.e., in Xcode 10.2 or above).

I think it’s reasonable to make this a 12.2-or-above check, even if that means the developer has to upgrade the SDK when iOS 12.3 comes out fixing the issue. On the other hand, if we limit this check to iOS 12.2 and a fix doesn’t come in iOS 12.3, then the developer has to upgrade the SDK anyways to avoid a regression.

Reply:
Yes, good catch - this should be a runtime check.